webpack/JobInvocationDetail/JobInvocationSystemStatusChart.js

```diff
@@ -0,0 +1,143 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { translate as __ } from 'foremanReact/common/I18n';
+import {
+  ChartDonut,
+  ChartLabel,
+  ChartLegend,
+  ChartTooltip,
+} from '@patternfly/react-charts';
+import {
+  DescriptionList,
+  DescriptionListTerm,
+  DescriptionListGroup,
+  DescriptionListDescription,
+  FlexItem,
+  Text,
+} from '@patternfly/react-core';
+import {
+  global_palette_green_500 as successedColor,
+  global_palette_red_100 as failedColor,
+  global_palette_blue_300 as inProgressColor,
+  global_palette_black_600 as canceledColor,
+} from '@patternfly/react-tokens';
+import DefaultLoaderEmptyState from 'foremanReact/components/HostDetails/DetailsCard/DefaultLoaderEmptyState';
+import './JobInvocationDetail.scss';
+
+const JobInvocationSystemStatusChart = ({
+  data,
+  isAlreadyStarted,
+  formattedStartDate,
+}) => {
+  const {
+    succeeded,
+    failed,
+    pending,
+    cancelled,
+    total,
+    total_hosts: totalHosts, // includes scheduled
+  } = data;
+  const chartData = [
+    { title: __('Succeeded:'), count: succeeded, color: successedColor.value },
+    { title: __('Failed:'), count: failed, color: failedColor.value },
+    { title: __('In Progress:'), count: pending, color: inProgressColor.value },
+    { title: __('Canceled:'), count: cancelled, color: canceledColor.value },
+  ];
+  const chartDonutTitle = () => {
+    if (total > 0) return `${succeeded.toString()}/${total}`;
+    if (totalHosts > 0) return `0/${totalHosts}`;
+    return '0';
+  };
+
+  const [legendWidth, setLegendWidth] = useState(270);
+
+  useEffect(() => {
+    const legendContainer = document.querySelector('.chart-legend');
```

Comment:
I understand it today, but probably would be best to add a comment on why its needed for future us :)